core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionManagerWrapper.java

```diff
@@ -0,0 +1,114 @@
+package org.pentaho.di.connections.vfs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.pentaho.di.connections.ConnectionDetails;
+import org.pentaho.di.connections.ConnectionManager;
+import org.pentaho.di.connections.ConnectionProvider;
+import org.pentaho.di.connections.vfs.provider.ConnectionFileName;
+import org.pentaho.di.core.bowl.Bowl;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.vfs.IKettleVFS;
+
+import java.util.List;
+
+/**
+ * The {@code VFSConnectionManagerWrapper} interface contains functionality related with VFS connections.
+ */
+public interface VFSConnectionManagerWrapper {
+
+  @NonNull
+  Bowl getBowl();
+
+  @NonNull
+  ConnectionManager getManager();
+
+  @NonNull
+  IKettleVFS getKettleVFS();
+
+  @NonNull
+  List<ConnectionManager.Type> getTypes();
+
+  @NonNull
+  List<String> getNames();
+
+  @Nullable <T extends VFSConnectionDetails> T getDetails( @Nullable String name );
+
+  @NonNull <T extends VFSConnectionDetails> T getExistingDetails( @Nullable String name ) throws KettleException;
+
+  @NonNull <T extends VFSConnectionDetails> VFSConnectionProvider<T> getProvider( T details );
+
+  @NonNull <T extends VFSConnectionDetails> VFSConnectionFileNameTransformer<T> getFileNameTransformer( T details );
+
+  /**
+   * Tests if a VFS connection is valid with default testing options.
+   * <p>
+   * This method delegates to {@link #test(VFSConnectionDetails, VFSConnectionTestOptions)} with a {@code null}
+   * {@code options} argument.
+   *
+   * @param details The VFS connection.
+   * @return {@code true} if the provided rootPath is valid; {@code false} otherwise.
+   */
+  <T extends VFSConnectionDetails> boolean test( @NonNull T details ) throws KettleException;
+
+  /**
```

Comment:
The methods `test`, `getResolvedRootPath` and `useBuckets` were moved here from `VFSConnectionProvider`.